pkg/auth/context_config.go

```diff
@@ -0,0 +1,25 @@
+package auth
+
+import (
+	"github.com/spf13/pflag"
+)
+
+type ContextConfig struct {
+}
+
+func NewContextConfig() *ContextConfig {
+	return &ContextConfig{}
+}
+
+func (c *ContextConfig) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&tenantUsernameClaim, "tenant-username-claim", tenantUsernameClaim, "Token claims key to retrieve the corresponding user principal.")
```

Comment:
Could we have a little bit more specific names for the flags?

The concept of "tenant" seems very broad. We are dealing with a tenant of what?

Reply:
The purpose of naming the key only "tenant" is to make it broad. If someone from the community wants to use the services they can configure their own claim keys and use the configuration to set them in the service. Same claims they can configure in the IDP. We have a lot of other inner dependencies which we still need to change with alternative opensource implementation. but this can be one step in that direction. It will be confusing for us as we don't have a tenant concept. but considering org ~ tenant